engine/config.h

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include <nlohmann/json.hpp>
+
+#include <optional>
+
+namespace pong
+{
+
+class Config
+{
+private:
+    nlohmann::json mConfig {};
+
+public:
+    static Config& GetInstance();
+    static bool LoadConfig(const std::string& filePath);
+
+    static std::optional<nlohmann::json> GetJsonValue(const std::string& key);
+
+    template<typename T>
+    static T GetValue(const std::string& key)
+    {
+        if (!GetInstance().mConfig.contains(key))
+        {
+            return T{};
+        }
+
+        try
+        {
+            return GetInstance().mConfig[key].get<T>();
+        }
+        catch(const std::exception&)
```

Comment:
as a mental exercise, how would you do this without exceptions? Many codebases disallow them, so always good to know both ways

Reply:
I suppose you could use specilization template functions to implement this function for each type. Then use the json's libraries own type checking ex  `mConfig[key].is_bool()` to see if it is good to return it. That would be my first approach at least.

Reply:
Seems feasible